### ISSUE TITLE
UITableViewAutomaticDimension layout issue for Header Views

### ISSUE THREAD

**Issue by msege** [fuzz-productions/FZAccordionTableView#27]
When i dont want to use delegate of tableview which sets height of section height, there is a problem about refresh issue. After several touches on sections, cell of related section is breaking down(ui problem). i guess it is about layoutifneeded after expand the section

**Comment by kgaidis** [comment]
Hi there,

Have you defined:
`func tableView(tableView: UITableView, estimatedHeightForHeaderInSection section: Int) -> CGFloat` ?
If that isn't defined I've seen some issues with layout before.

Otherwise, do you have exact steps how to reproduce the issue?

**Comment by vlade412** [comment]
Hi,

I have the same issue when I implement the UITableViewAutomaticDimension with autolayout. I added the methods:
- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section
- (CGFloat)tableView:(UITableView *)tableView estimatedHeightForRowAtIndexPath:(NSIndexPath *)indexPath
- (CGFloat)tableView:(UITableView *)tableView estimatedHeightForHeaderInSection:(NSInteger)section

but when I expand a section, the next Exception appear:

**\* Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '**\* -[__NSArrayM insertObject:atIndex:]: object cannot be nil'.

If I put a Exception Breakpoint, is stopped in the 334 line (endUpdates method). I think too that is a refresh issue. Any suggestions?

**Comment by kgaidis** [comment]
@vlade412 

I assume you are putting `UITableViewAutomaticDimension` ONLY for `heightForRowAtIndexPath`? Or for also the `heightForHeaderInSection`? I am not sure if headers support it. 

Also, if you change everything back to real values (say, 100 px), it does not crash?

**Comment by vlade412** [comment]
@kgaidis 

Yes, I worked before with UITableViewAutomaticDimension for heightForHeaderInSection, and work fine. If I put a value (no automatic dimension) it does not crash. I will calculate the height for header with code and not use autolayout to avoid exceptions. Thanks!

**Comment by kgaidis** [comment]
@vlade412 

Found this (a completely unrelated project having a similar issue):
http://stackoverflow.com/questions/38477121/uitableviewautomaticdimension-for-cell-or-header-view-height-causes-crash-on-uit

Also, from what I researched, using `UITableViewAutomaticDimension` for headers is not documented, so the state of it is unknown to me.

I'd be interested in seeing the full stack trace. 
Nonetheless, I should create a test project that uses `UITableViewAutomaticDimension` to see how reliable it is to use it with `FZAccordionTableView`. Cells should work, that is highly documented. Headers, not 100% sure.
